invenio_communities/models.py

```diff
@@ -228,6 +261,12 @@ def add_record(self, record):
             record[communities_key] = record[communities_key] + [self.id, ]
         else:
             record[communities_key] = [self.id, ]
+        if current_app.config["COMMUNITIES_OAI_ENABLED"] and "_oai" in record:
+            oai_spec = get_oaiset_spec(self.id)
```

Comment:
inveniosoftware/invenio-oaiserver#62 has been merged so you can write something like:
```python
oaiset = OAISet.query.get(get_oaiset_spec(self.id))
if oaiset is None:
    # create one or log error
else:
    oaiset.add_record(record)
```